content/blog/2019-09-23-grpc-on-dotnetcore.md

```diff
@@ -0,0 +1,50 @@
+---
+author: Sourabh Shirhatti
+author-link: https://twitter.com/sshirhatti
+date: "2019-09-23T00:00:00Z"
+published: true
+title: .NET Core ❤ gRPC
+url: blog/grpc-on-dotnetcore
+---
+
+_This is a guest post by [Sourabh Shirhatti](https://twitter.com/sshirhatti), a Program Manger on the .NET team at Microsoft._
+
+The .NET team at Microsoft has been working in close collaboration with the gRPC team since November 2018 on a new fully managed implementation of gRPC for .NET Core.
+
+We're pleased to announce that **grpc-dotnet** is now available with .NET Core 3.0 today!
+
+## How to get it?
+
+**grpc-dotnet** packages have just been released to [NuGet.org](https://www.nuget.org/profiles/grpc-packages) and are already available for use in your projects. These packages also require the latest .NET Core 3.0 shared framework. You can download the .NET Core 3.0 SDK for your dev machine and build servers from the [.NET Core 3.0 download page](https://aka.ms/netcore3download) to acquire the shared framework. 
+
+## Getting Started
+
+gRPC templates are included as part of the .NET SDK. To get started, navigate to a console window after installing the SDK and run the following commands.
+
+```sh
+dotnet new grpc -o GrpcGreeter
+cd GrpcGreeter
+dotnet run
+```
+
+To create a gRPC client and test with the newly created gRPC Greeter service, you can [follow the rest of this tutorial here](https://docs.microsoft.com/aspnet/core/tutorials/grpc/grpc-start).
+
+##  What's new?
+
+Unlike the existing C-Core based implementation ([Grpc.Core](https://github.com/grpc/grpc/tree/master/src/csharp)), the new libraries ([grpc-dotnet](https://github.com/grpc/grpc-dotnet)) make use of the existing networking primitives in the .NET Core Base Class Libraries (BCL). The diagram below highlights the difference between the existing **Grpc.Core** library and the new **grpc-dotnet** libraries.
+
+<p><img src="/img/grpc-dotnet.svg" alt="gRPC .NET Stack" style="max-width: 800px" /></p>
+
+On the server side, the `Grpc.AspNetCore.Server` package integrates into ASP.NET Core, allowing developers to benefit from ecosystem of common cross-cutting concerns of logging, configuration, dependency injection, authentication, authorization etc. which have already been solved by ASP.NET Core. Popular libraries in the ASP.NET ecosystem such as [Entity Framework Core (ORM)](https://github.com/aspnet/EntityFrameworkCore), [Serilog (Logging library)](https://github.com/serilog/serilog), and [Identity Server](https://github.com/IdentityServer/IdentityServer4) among others now work seamlessly with gRPC.
```

Comment:
+1 but we should also mention  what's going to be the relationship of both implementations?  

I've tried to cover that in these paragraphs:
https://github.com/grpc/grpc/tree/master/src/csharp#grpc-c

Also, some key points are mentioned here
https://github.com/grpc/grpc-dotnet#the-plan (the sections probably needs to get updated, but it's still mostly true).

We should probably provide a comparison table of both implementations because so far that's not something we've documented (e.g. grpc-dotnet is a winner in terms of smooth integration with ASP.NET Core but it's missing some features that Grpc.Core provides and also requires .NET Core 3 etc.). I think an accurate picture of the reality is that while grpc-dotnet is definitely a big step forward and something to be really excited about, there are pros and cons associated with the both implementations. That's also why we are providing the compatible API to allow users to flow between Grpc.Core and grpc-dotnet. 

One option is to come up with the comparison doc separately and make this blog post link to comparison of Grpc.Core vs grpc-dotnet (btw, that doc is something that should be updated continously as we implement new features, so it probably should be separate from this blogpost).

Reply:
I agree with your assessment that the comparison doc should be a living document that's frequently updated. Let's do that separately and leave this blog post as a point in time document.

I did however really like the verbiage you've used around introducing the two different implementations and I'll try and incorporate some of that in here

Reply:
@jtattermusch Do you prefer this?

Reply:
@shirhatti  looking pretty good now. Thanks!